pkgs/test_api/lib/src/backend/test_location.dart

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// The location of a test or group.
+class TestLocation {
+  final Uri uri;
+  final int line;
+  final int column;
+
+  TestLocation(String uri, this.line, this.column) : uri = Uri.parse(uri);
```

Comment:
If we want a Uri, I would be inclined to just take a Uri here.

Otherwise you end up in situations where people have a Uri, and they have to call `toString()` before passing it to you, just for you to parse it again. And parsing Uris is actually pretty expensive.

Reply:
Done! When I was writing the tests, it felt weird to have to put `Uri.parse()`/`Uri.file()` repeatedly, but ofc it's likely the caller is some test framework and the user isn't writing these explicitly, so that was a bad optimisation.